Lesson18/elifbasak/recipeBook.js

```diff
@@ -0,0 +1,336 @@
+  console.log("");
+  console.log("");
+}
+
+displayAllRecipes();
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Recipe
+-----------------------------------------------------------
+Function: addRecipe(name, ingredients, cookingTime)
+- Checks if a recipe with the same name exists.
+- If yes, log a warning and return.
+- If not, add the new recipe and log success.
+- ingredients should be an array like ['egg', 'milk', 'flour']
+*/
+
+function addRecipe(name, ingredients, cookingTime) {
+  if (
+    typeof name !== "string" ||
+    !Array.isArray(ingredients) ||
+    typeof cookingTime !== "number"
+  ) {
+    console.log(
+      `Please provide a string format for "name", an array for "ingredients", and a number for "cookingTime".`
+    );
+    console.log("");
+    console.log("");
+    return;
+  }
+  for (let i = 0; i < recipes.length; i++) {
+    if (name.toLocaleLowerCase() === recipes[i].name.toLocaleLowerCase()) {
+      console.warn(`There is already a recipe named ${name}.`);
+      console.log("");
+      console.log("");
+      return;
+    }
+  }
+  recipes.push({ name, ingredients, cookingTime });
+  console.log("Recipe added successfully.");
+  console.log(`Finished adding ${name} recipe.`);
+  console.log("");
+  console.log("");
+}
+
+addRecipe("sarma", ["rice", "onion", "tomato"], 90);
+addRecipe("Mantı", ["flour", "water", "ground beef", "onion", "yoghurt"], 120);
+addRecipe(90, "sarma", ["rice", "onion", "tomato"]);
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Recipe by Name
+-----------------------------------------------------------
+Function: viewRecipe(name)
+- Looks for the recipe by name and logs all its info.
+- If not found, shows a message.
+*/
+
+function viewRecipe(name) {
+  if (typeof name !== "string") {
+    console.log(`Please provide a string format for "name"`);
+    console.log("");
+    console.log("");
+    return;
+  }
+
+  console.log(`Looking for ${name} in the recipe book...`);
+  for (let i = 0; i < recipes.length; i++) {
+    if (name.toLocaleLowerCase() === recipes[i].name.toLocaleLowerCase()) {
+      console.log(`
+        Recipe Name: ${recipes[i].name}, 
+        Ingredients: ${recipes[i].ingredients}, 
+        Cooking Time: ${recipes[i].cookingTime} minutes.`);
+      console.log("");
+      console.log("");
+      return;
+    }
+  }
+  console.log(`There is no recipe named ${name}.`);
+  console.log("");
+  console.log("");
+}
+
+viewRecipe("Red Lentil Soup");
+viewRecipe("SARMA");
+viewRecipe("Roasted Vegetables");
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Recipe
+-----------------------------------------------------------
+Function: updateRecipe(name, newIngredients, newCookingTime)
+- Finds the recipe by name.
+- Updates ingredients and cookingTime.
+- Logs success or error message.
+*/
+
+function updateRecipe(name, newIngredients, newCookingTime) {
+  if (
```

Comment:
Possible improvement:
You have to repeat this data validation statement in many functions - create a utility function that just checks provided data types. Then, just call that function so you don't have to repeat the entire if statement everywhere.